lib/client.js

```diff
@@ -217,9 +217,10 @@ class Client extends EventEmitter {
       // TODO move this[kCallbacks] from being an array. The array allocation
       // is showing up in the flamegraph.
       const cb = this[kCallbacks].shift()
+      const request = this[kRequests].shift()
       this._needHeaders--
       this._lastBody = new Readable({ read: this[kRead].bind(this) })
-      this._lastBody.push = this[kRequests].shift().wrapSimple(this._lastBody, this._lastBody.push)
+      this._lastBody.push = request.wrapSimple(this._lastBody, this._lastBody.push)
```

Comment:
assigning `_lastBody` should be skipped if it's a `HEAD` because it has no body. I _think_ you might need to call `this[kRead]` anyway to kickstart reading the next response.

Reply:
I did skip the assignment, however the problem could have been the parser.
It wasn't calling kOnMessageComplete and kOnHeadersComplete. By changing the `skipBody` value to `1` or `true` it seems fixed.

See https://github.com/mcollina/undici/pull/9#discussion_r367902840